Sources/Public/Animation/AnimationView.swift

```diff
@@ -939,6 +955,87 @@ final public class AnimationView: AnimationViewBase {
     currentFrame = CGFloat(animation.startFrame)
   }
 
+  fileprivate func makeMainThreadAnimationLayer(for animation: Animation) -> MainThreadAnimationLayer {
+    MainThreadAnimationLayer(
+      animation: animation,
+      imageProvider: imageProvider,
+      textProvider: textProvider,
+      fontProvider: fontProvider)
+  }
+
+  fileprivate func makeCoreAnimationLayer(for animation: Animation) -> CoreAnimationLayer? {
+    do {
+      let coreAnimationLayer = try CoreAnimationLayer(
+        animation: animation,
+        imageProvider: imageProvider,
+        fontProvider: fontProvider,
+        compatibilityTrackerMode: .track)
+
+      coreAnimationLayer.didSetUpAnimation = { compatibilityIssues in
+        LottieLogger.shared.assert(
+          compatibilityIssues.isEmpty,
+          compatibilityIssues.compatibilityMessage)
+      }
+
+      return coreAnimationLayer
+    } catch {
+      LottieLogger.shared.assertionFailure("Encountered unexpected error \(error)")
+      return nil
+    }
+  }
+
+  fileprivate func makeAutomaticEngineLayer(for animation: Animation) -> CoreAnimationLayer? {
+    do {
+      // Attempt to set up the Core Animation layer. This can either throw immediately in `init`,
+      // or throw an error later in `CALayer.display()` that will be reported in `didSetUpAnimation`.
+      let coreAnimationLayer = try CoreAnimationLayer(
+        animation: animation,
+        imageProvider: imageProvider,
+        fontProvider: fontProvider,
+        compatibilityTrackerMode: .abort)
+
+      coreAnimationLayer.didSetUpAnimation = { [weak self] issues in
+        self?.automaticEngineLayerDidSetUpAnimation(issues)
+      }
+
+      return coreAnimationLayer
+    } catch {
+      if case CompatibilityTracker.Error.encounteredCompatibilityIssue(let compatibilityIssue) = error {
+        automaticEngineLayerDidSetUpAnimation([compatibilityIssue])
+      } else {
+        automaticEngineLayerDidSetUpAnimation([])
+      }
+
+      return nil
+    }
+  }
+
+  // Handles any compatibility issues with the Core Animation engine
+  // by falling back to the Main Thread engine
+  fileprivate func automaticEngineLayerDidSetUpAnimation(_ compatibilityIssues: [CompatibilityIssue]) {
+    // If there weren't any compatibility issues, then there's nothing else to do
+    if compatibilityIssues.isEmpty {
+      return
+    }
+
+    LottieLogger.shared.warn(
+      compatibilityIssues.compatibilityMessage
```

Comment:
Maybe we could include information that this is the first error, but there may be more (since we're using the `abort` mode)? 

And maybe details on the fact that this fallback has performance overhead, and you should specify the main thread rendering engine directly if you're seeing degraded performance

Reply:
good idea